tests/cmds/cmd_test_utils.py

```diff
@@ -0,0 +1,208 @@
+        if client_type == FarmerRpcClient:
+            return cast(FarmerRpcClient, self.farmer_rpc_client)  # type: ignore[return-value]
+        elif client_type == WalletRpcClient:
+            return cast(WalletRpcClient, self.wallet_rpc_client)  # type: ignore[return-value]
+        elif client_type == FullNodeRpcClient:
+            return cast(FullNodeRpcClient, self.full_node_rpc_client)  # type: ignore[return-value]
+        elif client_type == DataLayerRpcClient:
+            return cast(DataLayerRpcClient, self.data_layer_rpc_client)  # type: ignore[return-value]
+        elif client_type == SimulatorFullNodeRpcClient:
+            return cast(SimulatorFullNodeRpcClient, self.simulator_full_node_rpc_client)  # type: ignore[return-value]
+        else:
+            raise ValueError(f"Invalid client type requested: {client_type.__name__}")
+
+
+def create_service_and_wallet_client_generators(test_rpc_clients: TestRpcClients, default_root: Path) -> None:
+    # custom generators designed for testing
+
+    @asynccontextmanager
+    async def test_get_any_service_client(
+        client_type: Type[_T_RpcClient],
+        rpc_port: Optional[int] = None,
+        root_path: Path = default_root,
+        consume_errors: bool = True,
+    ) -> AsyncIterator[Tuple[_T_RpcClient, Dict[str, Any]]]:
+        node_type = node_config_section_names.get(client_type)
+        if node_type is None:
+            # Click already checks this, so this should never happen
+            raise ValueError(f"Invalid client type requested: {client_type.__name__}")
+        # load variables from config file
+        config = load_config(
+            root_path,
+            "config.yaml",
+            fill_missing_services=issubclass(client_type, DataLayerRpcClient),
+        )
+        self_hostname = config["self_hostname"]
+        if rpc_port is None:
+            rpc_port = config[node_type]["rpc_port"]
+        test_rpc_client = test_rpc_clients.get_client(client_type)
+
+        await test_rpc_client.create(self_hostname, uint16(rpc_port), root_path, config)
+        yield test_rpc_client, config
+
+    @asynccontextmanager
+    async def test_get_wallet_client(
+        wallet_rpc_port: Optional[int] = None,
+        fingerprint: Optional[int] = None,
+        root_path: Path = default_root,
+    ) -> AsyncIterator[Tuple[WalletRpcClient, int, Dict[str, Any]]]:
+        async with test_get_any_service_client(WalletRpcClient, wallet_rpc_port, root_path) as (wallet_client, config):
+            wallet_client.fingerprint = fingerprint  # type: ignore
+            assert fingerprint is not None
+            yield wallet_client, fingerprint, config
+
+    # override the functions
+    chia.cmds.cmds_util.get_any_service_client = test_get_any_service_client
+    chia.cmds.wallet_funcs.get_wallet_client = test_get_wallet_client  # type: ignore[attr-defined]
+
+
+def run_cli_command(capsys: Any, chia_root: Path, command_list: List[str]) -> Tuple[bool, str]:
```

Comment:
is `capsys` really `Any`? If we truly don't know what the type is, I'm under the impression `object` is better

Reply:
capsys is a defined in the _pytest module so i assumed it was not supposed to be imported, i can change it to that spooky type or object, what would you perfer?

Reply:
changed to object

Reply:
I see it's changed to `object` in one place, but not here